pkg/controller/daemon/daemoncontroller.go

```diff
@@ -1180,18 +1183,24 @@ func NewPod(ds *extensions.DaemonSet, nodeName string) *v1.Pod {
 
 // Predicates checks if a DaemonSet's pod can be scheduled on a node using GeneralPredicates
 // and PodToleratesNodeTaints predicate
-func Predicates(pod *v1.Pod, nodeInfo *schedulercache.NodeInfo) (bool, []algorithm.PredicateFailureReason, error) {
+func Predicates(pod *v1.Pod, nodeInfo *schedulercache.NodeInfo) (bool, []algorithm.PredicateFailureReason, bool, error) {
 	var predicateFails []algorithm.PredicateFailureReason
 	critical := utilfeature.DefaultFeatureGate.Enabled(features.ExperimentalCriticalPodAnnotation) && kubelettypes.IsCriticalPod(pod)
 
 	fit, reasons, err := predicates.PodToleratesNodeTaints(pod, nil, nodeInfo)
 	if err != nil {
-		return false, predicateFails, err
+		return false, predicateFails, false, err
 	}
 	if !fit {
 		predicateFails = append(predicateFails, reasons...)
 	}
 
+	fit, _, err = predicates.PodToleratesNodeNoExecuteTaints(pod, nil, nodeInfo)
```

Comment:
I see, here is mainly to check if want to evict the pod or not, but this seems a bit strange: In above `PodToleratesNodeTaints`, it was already checked both taints, and here we need to check the `NoExecute` taint separately.

I think that @janetkuo 's proposal is better and would involve less code change.